fedimint-client/src/sm/executor.rs

```diff
@@ -45,6 +47,32 @@ enum ExecutorDbPrefixes {
     InactiveStates = 0xa2,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct StateMachineCreated {
+    operation_id: OperationId,
+    module_id: ModuleInstanceId,
+}
+
+impl Event for StateMachineCreated {
+    const MODULE: Option<fedimint_core::core::ModuleKind> = None;
+
+    const KIND: EventKind = EventKind::from_static("sm-created");
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct StateMachineUpdated {
+    operation_id: OperationId,
+    terminal: bool,
+    outcome: serde_json::Value,
+    module_id: ModuleInstanceId,
+}
+
+impl Event for StateMachineUpdated {
+    const MODULE: Option<fedimint_core::core::ModuleKind> = None;
+
+    const KIND: EventKind = EventKind::from_static("sm-updated");
+}
+
```

Comment:
You probably have kind  of a "Roadmap" in mind for migrating away from the old notification mechanism we use for update streams rn, would be great to outline it in a tracking issue somewhere. We are introducing redundancy here that we should track and try to get rid of sooner rather than later.

Reply:
@elsirion we can continue in https://github.com/fedimint/fedimint/issues/6090